### PR TITLE
Feature sum

### PR DESCRIPTION
**Pull request by DavidVassett**

### CODE REVIEW

File: client/js/array-util.js [hunk 1]
```diff
@@ -10,7 +10,20 @@ const getLetterRange = function (firstLetter = 'A', numLetters) {
     .map(charCode => String.fromCharCode(charCode))
 }
 
+const calculateSum = function (arr) {
+  const result = []
```
Comment (speschellk) [inline client/js/array-util.js]: Name arrays after the items they contain. What is "result"? What kind of info will it hold? Arrays end with "s" in most cases (e.g. "columnSums" or "animals").

File: client/js/array-util.js [hunk 2]
```diff
@@ -10,7 +10,20 @@ const getLetterRange = function (firstLetter = 'A', numLetters) {
...
+  for (let i = 0; i < arr.length; i++) {
+    const sum = arr[i]
+      .map(val => parseInt(val))
+      .filter(val => !isNaN(val))
+      .reduce((a, b) => a + b, 0)
+    result.push(sum || '0')
```
Comment (speschellk) [inline client/js/array-util.js]: Always use semicolons to mark the end of an expression.

File: client/js/table-view.js [hunk 3]
```diff
@@ -72,6 +74,30 @@ class TableView {
     this.sheetBodyElement.appendChild(fragment)
   }
 
+  getColumnValues () {
+    const data = this.model.data
+    const numCols = this.model.numCols
+    const numRows = this.model.numRows
+    const result = []
```
Comment (speschellk) [inline client/js/table-view.js]: Same thing here. Now not only do I not know what "result" contains, but I've seen it twice in functions with different purposes.

File: client/js/table-view.js [hunk 4]
```diff
@@ -72,6 +74,30 @@ class TableView {
...
+    for (let i = 0; i < numCols; i++) {
+      const colValues = []
+      for (let j = 0; j < numRows; j++) {
+        if (data[i + ':' + j]) {
```
Comment (speschellk) [inline client/js/table-view.js]: Perfect place to use a template literal ( `` ) instead of doing string interpolation.

File: client/js/test/array-util.test.js [hunk 5]
```diff
@@ -24,8 +24,22 @@ describe('array-util', () => {
       expect(getLetterRange('B', 5)).toEqual(['B', 'C', 'D', 'E', 'F'])
     })
 
-    it('produces a valid single letter range ', () => {
+    it('produces a valid single letter range', () => {
       expect(getLetterRange('A', 1)).toEqual(['A'])
     })
   })
+
+  describe('calculateSum', () => {
```
Comment (speschellk) [inline client/js/test/array-util.test.js]: Add tests for more harmful edge cases. What if someone enters a name instead of a number? What about an empty string?

**Comment by DavidVassett** [comment]
Thanks for the review @speschellk. I have committed the requested changes.
